spec/models/spree/return_athorization_spec.rb

```diff
@@ -0,0 +1,180 @@
+        let(:shipwire_id)     { pending_order }
+        let(:return_items)    { [return_item] }
+
+        before do
+          return_authorization.return_items = return_items
+          return_authorization.order.update_attribute(:shipwire_id, shipwire_id)
+        end
+
+        it 'post to shipwire' do
+          expect(return_authorization).to receive(:process_shipwire_return!).and_call_original
+
+          return_authorization.save
+        end
+
+        it 'add an error message' do
+          return_authorization.save
+
+          expect(return_authorization.errors.messages).to have_key(:shipwire_unprocessed)
+          expect(return_authorization.errors.messages[:shipwire_unprocessed].first).to eq("Only orders that are \"processed\" and not \"cancelled\" can be returned")
+        end
+      end
+
+      context 'when a return for the order was already been reported',
+              vcr: { cassette_name: 'spree/return_authorization_entity_exists' } do
+
+        let(:inventory_unit)  { build(:inventory_unit, variant: variant_1) }
+        let(:return_item)     { build(:return_item, inventory_unit: inventory_unit) }
+        let(:shipwire_id)     { already_returned_order }
+        let(:return_items)    { [return_item] }
+
+        before do
+          return_authorization.return_items = return_items
+          return_authorization.order.update_attribute(:shipwire_id, shipwire_id)
+        end
+
+        it 'add an error message' do
+          return_authorization.save
+
+          expect(return_authorization.errors.messages).to have_key(:shipwire_already_reported)
+          expect(return_authorization.errors.messages[:shipwire_already_reported].first).to eq('You have already reported this issue.')
+        end
+      end
+    end
+
+    context 'when shipwire does accept the return' do
+      context 'when return has 1 return item',
+              vcr: { cassette_name: 'spree/return_authorization_create_entity_single_return_item' } do
+
+        let(:inventory_unit)  { build(:inventory_unit, variant: variant_1) }
+        let(:return_item)     { build(:return_item, inventory_unit: inventory_unit) }
+        let(:shipwire_id)     { shipped_order_single_return_item }
+        let(:return_items)    { [return_item] }
+        let(:items_array)     { [{ sku: variant_1.sku, quantity: 1 }] }
+
+        before do
+          return_authorization.return_items = return_items
+
+          return_authorization.order.shipments.first.ship
+          return_authorization.order.update_attribute(:shipwire_id, shipwire_id)
+        end
+
+        it 'post to shipwire' do
+          expect(return_authorization).to receive(:process_shipwire_return!).and_call_original
+          return_authorization.save
+          expect(return_authorization.shipwire_id).not_to be nil
+        end
+
+        it 'places a single item in the return' do
+          expect(return_authorization.to_shipwire[:items]).to eq items_array
+        end
```

Comment:
should we test that the `return_authorization` has the correct `shipwire_id` as well?